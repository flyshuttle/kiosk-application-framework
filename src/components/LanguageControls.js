import React from "react";

export default function LanguageControls(props) {
  const { locales, currentLocale, changeLanguage } = props;

  if (locales === undefined) {
    return "Locales prop for LanguageControls not set";
  }

  return (
    <div className="language-controls">
      {locales.map((locale, key) => {
        let className = "btn ";
        className =
          className +
          (currentLocale === locale.code
            ? "btn-primary"
            : "btn-outline-primary");

        return (
          <button
            key={key}
            className={className}
            onClick={() => changeLanguage(locale.code)}
          >
            {locale.name}
          </button>
        );
      })}
    </div>
  );
}
