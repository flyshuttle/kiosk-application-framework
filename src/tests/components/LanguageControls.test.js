import React from "react";
import ReactDOM from "react-dom";
import { create } from "react-test-renderer";

import LanguageControls from "../../components/LanguageControls";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<LanguageControls />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the component with no props", () => {
  const component = create(<LanguageControls />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders the component with one language", () => {
  const locales = [{ code: "en", name: "English" }];
  const component = create(<LanguageControls locales={locales} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders the component with two languages", () => {
  const locales = [
    { code: "en", name: "English" },
    { code: "fr", name: "french" }
  ];
  const component = create(<LanguageControls locales={locales} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
