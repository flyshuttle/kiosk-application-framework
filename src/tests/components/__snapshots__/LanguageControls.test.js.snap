// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component with no props 1`] = `"Locales prop for LanguageControls not set"`;

exports[`renders the component with one language 1`] = `
<div
  className="language-controls"
>
  <button
    className="btn btn-outline-primary"
    onClick={[Function]}
  >
    English
  </button>
</div>
`;

exports[`renders the component with two languages 1`] = `
<div
  className="language-controls"
>
  <button
    className="btn btn-outline-primary"
    onClick={[Function]}
  >
    English
  </button>
  <button
    className="btn btn-outline-primary"
    onClick={[Function]}
  >
    french
  </button>
</div>
`;
